api/src/main/java/org/apache/iceberg/expressions/ManifestEvaluator.java

```diff
@@ -329,5 +338,12 @@ public Boolean or(Boolean leftResult, Boolean rightResult) {
 
       return ROWS_MIGHT_MATCH;
     }
+
+    private boolean allValuesAreNull(PartitionFieldSummary summary) {
+      // containsNull encodes whether at least one partition value is null, lowerBound is null if all partition values
+      // are null; in case bounds don't include NaN value, containsNaN needs to be checked against.
+      return summary.containsNull() && summary.lowerBound() == null &&
+          summary.containsNaN() != null && !summary.containsNaN();
```

Comment:
`containsNaN` could only be true if the type is `float` or `double`. Should this check the column type before checking `containsNaN`? Then we would be able to ignore it for most cases. For example, the test below uses a string column but has no `containsNaN` values, so it doesn't catch that all values are null.

Reply:
Yes, sorry I forgot to address this comment... Will update